src/pages/index.js

```diff
@@ -63,48 +100,158 @@ Array.from(document.forms).forEach((formElement) => {
 });
 /* END DECLARATIVE SECTION */
 
-/* PROFILE SECTION */
+/* PROFILE INFO SECTION */
 const userProfile = new UserInfo({
+  // pass as arguments the HTML DOM profile elements
   name: profileName,
   description: profileDescription,
+  avatar: profileAvatar,
 });
 
-const popupProfile = new PopupWithForm(
+buttonEditProfile.addEventListener("click", () => {
+  const { name, description } = userProfile.getUserInfo();
+  inputProfileName.value = name;
+  inputProfileDescription.value = description;
+  formValidators["profile_info_form"].resetFormValidation(); // input always valid when taking from the html page
+  popupProfileInfo.open();
+});
+
+const popupProfileInfo = new PopupWithForm(
   modalWindowProfile,
   (event, { name, description }) => {
     event.preventDefault();
-    userProfile.setUserInfo(name, description);
-    popupProfile.close();
+    popupProfileInfo.setRender(true); // renders faster when synced before the async begins
+    api
+      .setUserProfileData(name, description)
+      .then(() => {
+        userProfile.setUserInfo(name, description);
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupProfileInfo.close();
+        popupProfileInfo.setRender(false);
+      });
   },
-  configuration.inputSelector
+  configuration.inputSelector,
+  submitProfileInfo,
+  "Saving..."
 );
-popupProfile.setEventListeners();
+popupProfileInfo.setEventListeners();
+/* END PROFILE INFO SECTION */
 
-buttonEditProfile.addEventListener("click", () => {
-  const { name, description } = userProfile.getUserInfo();
-  inputProfileName.value = name;
-  inputProfileDescription.value = description;
-  formValidators["profile_form"].resetFormValidation(); // input always valid when taking from the html page
-  popupProfile.open();
+/* PROFILE AVATAR SECTION */
+buttonUpdateAvatar.addEventListener("click", () => {
+  inputProfileAvatar.value = profileAvatar.src;
+  popupProfileAvatar.open();
 });
+const popupProfileAvatar = new PopupWithForm(
+  modalWindowProfileAvatar,
+  (event, { avatar }) => {
+    event.preventDefault();
+    popupProfileAvatar.setRender(true);
+    api
+      .setUserProfileAvatar(avatar)
+      .then(() => {
+        profileAvatar.src = avatar;
```

Comment:
`        profileAvatar.src = avatar;
`
`setAvatar` should be used here

`UserInfo` should set the entire profile data (including the **avatar**)